ttl/ttl.go

```diff
@@ -13,29 +13,40 @@ import (
 
 var ErrNotFound = errors.New("not found")
 
+// value is an opaque structure that wraps a value that is store in the TTL
+// key/value map.
 type value struct {
 	value any
 
-	expired func(any, any) // called when expired
-	remove  func(any, any) // called when removed
+	expired func(any, any) // called when TTL expires
+	remove  func(any, any) // called when removed from map
+
+	timeoutExpired bool // set when this value has expired
 
 	// Value context
 	ctx    context.Context
 	cancel context.CancelFunc
 }
 
+// TTL is an opaque structure that wraps the TTL key/value map.
```

Comment:
I think this comment could be made more descriptive. I don't think "wraps the TTL key/value map" really makes sense, because TTL is basically the key/value store from a user's perspective.